### ISSUE TITLE
Изменить работу команды firstboot

### ISSUE THREAD

**Issue by p0i5k** [OpenIPC/firmware#241]
Сейчас команда firstboot выполняет форматирование раздела rootfs_data.
Добавить:
- подготовку файла /etc/profiler.yaml для текущей платы данного производителя
- настройку пинов в /etc/majestic.yaml
- /etc/fw_env.config

**Comment by themactep** [comment]
Получается, поскольку firstboot выполняется на камере, исходные данные для формирования конфига профайлера тоже должны быть на камере. И на ней же будет содаваться конфиг профайлера. Мы из большого мешка будем копировать часть данных в маленький, плодя сущности и отъедая место? Рабочий конфиг под конкретное железо надо строить на этапе сборки прошивки, вне камеры, и сохранять в прошивку только его, а не тащить на камеру данные, которые к этой конретной железяке не относятся. Сейчас, например, в профайлере указаны все варианты gpio для всех процов всех вендоров, но когда прошивка уже на камере, ей нет дела до разных иных моделей. Эти данные просто занимают полезное место. Надо наоборот выносить все универсальное из готовой прошивки на этап сборки, а не переносить сборку на камеру.

**Comment by p0i5k** [comment]
Прошивка универсальна для процессора, а не производителя камеры. Поэтому в ней есть драйвера и для десятка сенсоров. Копирование - это плата за универсальность и установку в один клик.

**Comment by themactep** [comment]
О том и речь. Зачем тащить в камеру профайлер и базу по всеми GPIO по всем процам? Что мешает сформировать конфиг с пинами для конкретного процессора и записать его в /rom? И не будет нужды создавать его заново всякий раз при запуске firstboot.

**Comment by p0i5k** [comment]
Формирование конфига для конкретного процессора вынесено в https://github.com/OpenIPC/firmware/issues/240

**Comment by widgetii** [comment]
Если в firmware есть база данных по производителям плат (например, как у XM имя модели) с их детальной конфигурацией, то можно в env записывать отдельным параметром только имя и брать настройки после firstboot

**Comment by p0i5k** [comment]
Да, coupler сейчас уже устанавливается переменную manufacturer. Там, где OpenIPC уже установлен или установлен вручную, надо будет прописать его вручную.
Только тогда увеличится длина команды для чтения параметров на `manufacturer=${fw_print manufacturer}`

**Comment by openipc-ai** [comment]
Закрываю как неактуальное: тикет 2022 года, за три с лишним года к нему никто не возвращался, а `firstboot` с тех пор менялся не раз.

Если в текущем поведении команды что-то не так, заведите, пожалуйста, новый issue — с описанием того, что происходит сейчас, у него будет шанс найти исполнителя.
